File: materials-evaluation-spa/src/app/models/QualityProperty.ts
enum QualityType {
  Quantitative = 'Quantitative',
  Qualitative = 'Qualitative',
}

export interface QualityProperty {
  id: string;
  acronym: string;
  description: string;
  type: QualityType;

  // apenas para tipo Quantitative
  quantitativeDecimals: number;
  quantitativeUnit: string;
  quantitativeNominalValue: number;
  quantitativeInferiorLimit: number;
  quantitativeSuperiorLimit: number;
}

File: materials-evaluation-dotnet/Database/Models/QualityProperty.cs
namespace MaterialsEvaluation.Database;

public class QualityProperty
{
    public long Id { get; set; }
    public string? Acronym { get; set; }
    public string? Description { get; set; }
    public string? Type { get; set; }

    // Apenas para característica quantitativa
    // NOTE: Pode ter um jeito diferente de fazer isso, como um modelo especifico pra quantitativo?
    public int? QuantitativeDecimals { get; set; }
    public string? QuantitativeUnit { get; set; }
    public double? QuantitativeNominalValue { get; set; }
    public double? QuantitativeInferiorLimit { get; set; }
    public double? QuantitativeSuperiorLimit { get; set; }
}
